src/main/java/gift/service/MemberServiceImpl.java

```diff
@@ -0,0 +1,84 @@
+package gift.service;
+
+import gift.dto.MemberLogInRequestDto;
+import gift.dto.MemberLogInResponseDto;
+import gift.dto.MemberResponseDto;
+import gift.entity.Member;
+import gift.exception.InvalidPasswordException;
+import gift.exception.MemberEmailAlreadyExistsException;
+import gift.exception.MemberEmailNotExistsException;
+import gift.repository.MemberRepository;
+import gift.util.JwtProvider;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class MemberServiceImpl implements MemberService {
+    private final MemberRepository memberRepository;
+    private final JwtProvider jwtProvider;
+
+    public MemberServiceImpl(MemberRepository memberRepository, JwtProvider jwtProvider) {
+        this.memberRepository = memberRepository;
+        this.jwtProvider = jwtProvider;
+    }
+
+    @Override
+    public MemberLogInResponseDto registerMember(MemberLogInRequestDto requestDto) {
+        if (memberRepository.findMemberByEmail(requestDto.email()).isPresent()) {
+            throw new MemberEmailAlreadyExistsException();
+        }
+
+        Member member = new Member(
+            null,
+            requestDto.email(),
+            requestDto.password());
+        member = memberRepository.registerMember(member);
+
+        return new MemberLogInResponseDto(jwtProvider.createToken(member));
+    }
+
+    @Override
+    public MemberLogInResponseDto findMemberToLogIn(MemberLogInRequestDto requestDto) {
+        Optional<Member> memberOptional = memberRepository.findMemberByEmail(requestDto.email());
+        Member member = memberOptional.orElseThrow(MemberEmailNotExistsException::new);
+
+        if (!requestDto.password().equals(member.getPassword())) {
+            throw new InvalidPasswordException("비밀번호가 일치하지 않습니다.");
```

Comment:
```suggestion
            throw new InvalidPasswordException("이메일 또는 비밀번호가 일치하지 않습니다.");
```
- 보안과 관련한 예외 메시지는 구체적으로 작성하면 Brute Force 등의 공격을 허용하기 좋습니다.

Reply:
그렇겠네요! 수정하겠습니다 😊